src/map/utils/zoneutils.cpp

```diff
@@ -1209,6 +1213,16 @@ namespace zoneutils
         }
 
         luautils::AfterZoneIn(PChar);
+
+        if (PChar != nullptr && PChar->objtype == TYPE_PC)
```

Comment:
While I am here, I don't know why, but something tells me this logic, in general, doesn't belong here.

Characters het destryed and rebuilt each time they zone, because of reasons. This logic should be place there, while the character is being constructed, not after they already have.

Reply:
> Characters get destroyed and rebuilt each time they zone, because of reasons.
> This logic should be placed there, while the character is being constructed, not after they already have.

Can you point me where to look? I'm not really familiar with this codebase